ddtrace/context.py

```diff
@@ -29,6 +29,21 @@ def __init__(self):
         self._finished_spans = 0
         self._current_span = None
         self._lock = threading.Lock()
+        self._base_parent_span_id = None
+        self._base_parent_trace_id = None
+
+    def set_base_parent_span_ids(self, trace_id, span_id):
```

Comment:
While this behavior makes sense to prevent the "waterfall effect" where each concurrent span is child of the previous one, probably we need to add some tests for different use cases so that we're sure to cover:
* 1 task parent -> 1 concurrent task
* 1 task parent -> 5 concurrent tasks (they share the same parent)
* 1 task parent -> 5 concurrent tasks -> 1 task (the 5 concurrent tasks share the same parent, each one has one child that should not have the root task as a parent)

Later on, we may think about more test cases but for now it's a good start to see if we can improve this part later.